core/io/file_access_patched.cpp

```diff
@@ -0,0 +1,208 @@
+/**************************************************************************/
+/*                         This file is part of:                          */
+/*                             GODOT ENGINE                               */
+/*                        https://godotengine.org                         */
+/**************************************************************************/
+/* Copyright (c) 2014-present Godot Engine contributors (see AUTHORS.md). */
+/* Copyright (c) 2007-2014 Juan Linietsky, Ariel Manzur.                  */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "file_access_patched.h"
+
+#include "file_access_pack.h"
+
+#include "core/io/delta_encoding.h"
+#include "core/os/os.h"
+
+Error FileAccessPatched::_apply_patch() const {
+	ERR_FAIL_COND_V(!is_open(), FAILED);
+
+	String path = old_file->get_path();
+	Vector<PackedData::PackedFile> delta_patches = PackedData::get_singleton()->get_delta_patches(path);
+	Vector<uint8_t> old_file_data = old_file->get_buffer(old_file->get_length());
+
+	for (int i = 0; i < delta_patches.size(); ++i) {
+		const PackedData::PackedFile &delta_patch = delta_patches[i];
+		Error err = OK;
+
+		uint64_t total_usec_start = OS::get_singleton()->get_ticks_usec();
+		uint64_t io_usec_start = OS::get_singleton()->get_ticks_usec();
+
+		Ref<FileAccess> patch_file;
+
+		if (delta_patch.bundle) {
+			patch_file = FileAccess::open(delta_patch.pack, FileAccess::READ | FileAccess::SKIP_PACK, &err);
```

Comment:
That's not how `bundle` works, it is supposed to have file content outside `.pack` stored by its original name, which won't work for patches (since file name is the same for both base and patch).

I guess we can make some system for this, but I would just fail if `delta_patch.bundle` is set, since there's no condition when patch pack with `bundle` can be written.